pipeline-model-api/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/ast/ModelASTAgent.java

```diff
@@ -22,19 +23,49 @@ public ModelASTAgent(Object sourceLocation) {
     @Override
     public JSONObject toJSON() {
         final JSONObject j = new JSONObject();
-        j.accumulate("type", agentType.toJSON());
 
-        if (variables != null) {
-            if (variables instanceof ModelASTClosureMap &&
-                    !((ModelASTClosureMap) variables).getVariables().isEmpty()) {
-                j.accumulate("arguments", variables.toJSON());
-            } else if (variables instanceof ModelASTValue) {
-                j.accumulate("argument", variables.toJSON());
+        // Handle JENKINS-43016 - round-trip empty-string label agent into agent any.
+        if (isEmptyStringLabelAgent()) {
+            j.accumulate("type", "any");
+        } else {
+            j.accumulate("type", agentType.toJSON());
+
+            if (variables != null) {
+                if (variables instanceof ModelASTClosureMap &&
+                        !((ModelASTClosureMap) variables).getVariables().isEmpty()) {
+                    j.accumulate("arguments", variables.toJSON());
+                } else if (variables instanceof ModelASTValue) {
+                    j.accumulate("argument", variables.toJSON());
+                }
             }
         }
         return j;
     }
 
+    private boolean isEmptyStringLabelAgent() {
+        if (agentType.getKey().equals("label") || agentType.getKey().equals("node")) {
```

Comment:
I feel like I've missed some past large change, when was `agent node` introduced?

Reply:
Yeesh, it was a while ago. I can't honestly remember. It's just an alias for `label`, really.

Reply:
Yeah, I have some slither of memory about that now that you mention it :)